modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeCatchAllCheck.java

```diff
@@ -513,6 +521,112 @@ else if (!_newMessage) {
 		return newContent;
 	}
 
+	private String _formatGeneral(
+		String content, String fileName, JSONObject jsonObject) {
+
+		String newContent = content;
+
+		Pattern pattern = _getPattern(jsonObject);
+
+		Matcher matcher = pattern.matcher(content);
+
+		while (matcher.find()) {
+			String methodCall = matcher.group();
+
+			String from = jsonObject.getString("from");
+			String to = jsonObject.getString("to");
+
+			if (from.startsWith("regex:")) {
+				newContent = newContent.replaceAll(pattern.toString(), to);
+			}
+			else if (from.contains(StringPool.OPEN_PARENTHESIS)) {
+				newContent = _formatMethodCall(
+					fileName, from, newContent, jsonObject, matcher, newContent,
+					to);
+			}
+			else {
+				Set<String> keys = jsonObject.keySet();
+
+				if (!keys.contains("to")) {
+					addMessage(fileName, _getMessage(jsonObject));
+
+					_newMessage = true;
+
+					continue;
+				}
+
+				newContent = StringUtil.replaceFirst(
+					newContent, methodCall,
+					StringUtil.replace(methodCall, from, to), matcher.start());
+			}
+		}
+
+		if (!content.equals(newContent)) {
+			newContent = _addReplacementDependencies(
+				fileName, jsonObject, newContent);
+		}
+
+		return newContent;
+	}
+
+	private String _formatJava(
```

Comment:
I'm not sure this is a good name for the method, `_formatJava` seems to be general and we already have a `_formatGeneral`